lib/ansible/modules/cloud/amazon/elb_target.py

```diff
@@ -166,6 +167,7 @@ def describe_targets(connection, module, tg_arn, target):
                          exception=traceback.format_exc())
 
 
+@AWSRetry.jittered_backoff(retries=5, delay=5)
```

Comment:
ClientError is always caught and handled in this function (and the one below) so these decorators have no effect. AWSRetry is intended to be used before ClientError is handled. You should wrap the individual API calls that are problematic inside of the exception handling.

A lot of AWS modules do this by creating a smaller function to retry with the API call like this to then call inside of the exception handling:

```python
@AWSRetry.jittered_backoff(retries=5, delay=5)
def register_targets_with_backoff(connection, target, target_group_arn):
    connection.register_targets(TargetGroupArn=target_group_arn, Targets=[target])

def register_target(connection, module):
    ...
            try:
                register_targets_with_backoff(connection, target, target_group_arn)
                ...
            except ClientError as e:
    ...
```
Alternatively you could use the decorator inside of the exception handling:
```python
def register_target(connection, module):
    ...
            try:
                AWSRetry.jittered_backoff()(connection.register_targets)(TargetGroupArn=target_group_arn, Targets=[target])
                ...
            except ClientError as e:
    ...
```

There are additional details and examples here https://docs.ansible.com/ansible/latest/dev_guide/platforms/aws_guidelines.html#api-throttling-rate-limiting-and-pagination.

Reply:
fixed.